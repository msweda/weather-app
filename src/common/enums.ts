export enum RootRoute {
  Home = '/',
  Weather = '/weather',
  Forecast = '/forecast',
}

export enum AsyncStatus {
  Idle = 'IDLE',
  InProgress = 'IN_PROGRESS',
  Succeeded = 'SUCCEEDED',
  Failed = 'FAILED',
}
