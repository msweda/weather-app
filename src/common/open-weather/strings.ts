export const noGeolocation = 'Geolocation must be enabled to use this feature.'
export const weatherFailed =
  'Weather information is not available at this time.'
export const forecastFailed =
  'Forecast information is not available at this time.'
export const weatherIconUrl = (weatherIcon: string) =>
  `http://openweathermap.org/img/w/${weatherIcon}.png`
