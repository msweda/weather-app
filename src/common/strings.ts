export const appHeading = 'Matthew Sweda\'s Weather App';
export const phoneNumber = '(248)-770-7487';
export const email = 'msweda.developer@gmail.com';
export const location = 'Detroit, MI';
export const copyright = (timestamp: number) => `Â© Matthew Sweda ${new Date(timestamp).getFullYear()}`;
export const emailLink = `mailto:${email}`;
export const linkedInLink = 'https://www.linkedin.com/in/matthew-sweda-118570ba/';
export const instagramLink = 'https://www.instagram.com/mattsweda/';