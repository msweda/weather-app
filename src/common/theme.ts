// create palette here: https://material.io/color/

import { createMuiTheme } from '@material-ui/core/styles'

export default createMuiTheme({
  typography: {
    useNextVariants: true,
  },
  palette: {
    primary: {
      main: '#2d4671',
    },
    secondary: {
      main: '#6ba099',
    },
    background: {
      default: '#2d4671',
      paper: '#eee',
    },
    text: {
      primary: '#fff',
      secondary: '#000',
    },
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        body: {
          display: 'flex',
          minHeight: '100vh',
        },
        '#app': {
          flex: 1,
          display: 'flex',
          flexDirection: 'column',
        },
      },
    },
  },
})
