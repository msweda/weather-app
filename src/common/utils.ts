import createBrowserHistory from 'history/createBrowserHistory'

export const browserHistory = createBrowserHistory()

export function isGeolocationAvailable(): boolean {
  return 'geolocation' in navigator
}

export function getCurrentPosition() {
  return new Promise<Position>((resolve, reject) => {
    if (!isGeolocationAvailable()) {
      throw new Error('Geolocation is unavailable')
    }
    navigator.geolocation.getCurrentPosition(
      position => resolve(position),
      error => reject(error)
    )
  })
}
