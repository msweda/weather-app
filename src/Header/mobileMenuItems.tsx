import { RootRoute } from '@/common/enums'
import AcUnitIcon from '@material-ui/icons/AcUnit'
import Filter5Icon from '@material-ui/icons/Filter5'
import * as React from 'react'

interface MobileMenuItem {
  icon: JSX.Element
  label: string
  route: RootRoute
}

const mobileMenuItems: MobileMenuItem[] = [
  {
    icon: <AcUnitIcon />,
    label: 'Weather',
    route: RootRoute.Weather,
  },
  {
    icon: <Filter5Icon />,
    label: 'Forecast',
    route: RootRoute.Forecast,
  },
]

export default mobileMenuItems
