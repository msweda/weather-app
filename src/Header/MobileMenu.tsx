import { RootRoute } from '@/common/enums'
import { browserHistory } from '@/common/utils'
import Drawer from '@material-ui/core/Drawer'
import IconButton from '@material-ui/core/IconButton'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import ListItemIcon from '@material-ui/core/ListItemIcon'
import ListItemText from '@material-ui/core/ListItemText'
import {
  createStyles,
  Theme,
  WithStyles,
  withStyles,
} from '@material-ui/core/styles'
import MenuIcon from '@material-ui/icons/Menu'
import * as React from 'react'
import mobileMenuItems from './mobileMenuItems'

class MobileMenu extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      isOpen: false,
    }
  }

  render() {
    const { classes } = this.props
    return (
      <div>
        <IconButton onClick={this.handleOpen}>
          <MenuIcon className={classes.menuButton} />
        </IconButton>
        <Drawer
          anchor="right"
          open={this.state.isOpen}
          onClose={this.handleClose}
        >
          <List>
            {mobileMenuItems.map((item, index) => (
              <ListItem
                key={`mobileMenu-listItem-${index}`}
                button={true}
                onClick={this.handleMenuItemClick.bind(this, item.route)}
              >
                <ListItemIcon>{item.icon}</ListItemIcon>
                <ListItemText secondary={item.label} />
              </ListItem>
            ))}
          </List>
        </Drawer>
      </div>
    )
  }

  handleOpen = () => {
    this.setState({ isOpen: true })
  }

  handleClose = () => {
    this.setState({ isOpen: false })
  }

  handleMenuItemClick = (route: RootRoute) => {
    browserHistory.push(route)
    this.handleClose()
  }
}

const styles = (theme: Theme) =>
  createStyles({
    menuButton: {
      color: theme.palette.background.paper,
    },
  })

interface Props extends WithStyles<typeof styles> {}
interface State {
  isOpen: boolean
}

export default withStyles(styles)(MobileMenu)
