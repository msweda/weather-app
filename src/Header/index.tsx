import { appHeading } from '@/common/strings'
import { createStyles, WithStyles, withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import * as React from 'react'
import MobileMenu from './MobileMenu'

const Header: React.FC<Props> = ({ classes }) => (
  <div className={classes.header}>
    <Typography variant="h6">{appHeading}</Typography>
    <MobileMenu />
  </div>
)

const styles = () =>
  createStyles({
    header: {
      height: 50,
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-between',
      alignItems: 'center',
      padding: '5px 20px',
      boxShadow: '0 1px 2px rgba(0,0,0,0)',
    },
  })

interface Props extends WithStyles<typeof styles> {}

export default withStyles(styles)(Header)
