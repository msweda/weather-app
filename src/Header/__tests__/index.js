import { appHeading as expectedAppHeading } from '@/common/strings'
import { createShallow } from '@material-ui/core/test-utils'
import Typography from '@material-ui/core/Typography'
import * as React from 'react'
import Header from '../index'
import MobileMenu from '../MobileMenu'

describe('<Header />', () => {
  let shallow
  let wrapper

  beforeEach(() => {
    shallow = createShallow({ untilSelector: Header })
    wrapper = shallow(<Header />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the app heading', () => {
    const appHeading = wrapper
      .find(Typography)
      .children()
      .text()
    expect(appHeading).toEqual(expectedAppHeading)
  })

  it('should render <MobileMenu />', () => {
    expect(wrapper.find(MobileMenu).exists()).toBeTruthy()
  })
})
