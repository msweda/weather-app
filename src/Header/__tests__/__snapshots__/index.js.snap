// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should match snapshot 1`] = `
<div
  className="Header-header-1"
>
  <WithStyles(Typography)
    variant="h6"
  >
    Matt's Weather App
  </WithStyles(Typography)>
  <WithStyles(MobileMenu) />
</div>
`;
